SocietalConstructionTool/Runtime/Runtime.cs

```diff
@@ -5,7 +5,7 @@ public class Runtime() : IRuntime
         public void Run(IRuntimeContext initialContext)
         {
             IRuntimeContext prevCtx = initialContext;
-            while (!prevCtx.ShouldExit)
+            while (!prevCtx.ShouldExit && prevCtx.AgentHandler.Agents.Any())
```

Comment:
Should this logic maybe be encapsulated in `ShouldExit` instead, since it's already a property?
Maybe this make testing hard idk